dimod/generators/constraints.py

```diff
@@ -91,9 +92,8 @@ def combinations(n, k, strength=1, vartype=BINARY):
     lbias = float(strength*(1 - 2*k))
     qbias = float(2*strength)
 
-    bqm = BinaryQuadraticModel.empty(BINARY)
-    bqm.add_variables_from(((v, lbias) for v in variables))
-    bqm.add_interactions_from(((u, v, qbias) for u, v in itertools.combinations(variables, 2)))
-    bqm.offset += strength*(k**2)
+    Q = np.triu(np.ones((n,n))*qbias, k=1)
```

Comment:
`n` can also be a list or a set, not just int (see docstring above) -- in which case items define variable names (and `np.ones` above fails).

You could use `len(n)` instead of `n` above, and then relabel the bqm with:
```
bqm.relabel_variables(dict(zip(range(len(n)), n)))
```
(could also skip relabeling when user supplies int for n)

Reply:
Think I updated this - let me know if more changes are needed for this